<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>785</width>
    <height>475</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Генератор часових рядів</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="2">
     <widget class="QGroupBox" name="fileSelectionGroupBox">
      <property name="title">
       <string>Налаштування генератора:</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QRadioButton" name="chooseConfigRadioButton">
         <property name="text">
          <string>Конфігураційний файл</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="configPathLineEdit">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>middle_config.json</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="chooseConfigButton">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Огляд...</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QRadioButton" name="chooseDatasetRadioButton">
           <property name="text">
            <string>Датасет:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="chooseDatasetLineEdit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>datasets/workload_dataset/workload.csv</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="chooseDatasetButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Огляд...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QSpinBox" name="stepSpinBox">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>86400</number>
      </property>
      <property name="value">
       <number>60</number>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="startTimeLabel">
      <property name="text">
       <string>Дата та час початку:</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="stepLabel">
      <property name="text">
       <string>Часовий крок (секунди):</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="endTimeLabel">
      <property name="text">
       <string>Дата та час кінця:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QDateTimeEdit" name="startDateTimeEdit">
      <property name="dateTime">
       <datetime>
        <hour>0</hour>
        <minute>0</minute>
        <second>0</second>
        <year>2025</year>
        <month>5</month>
        <day>6</day>
       </datetime>
      </property>
      <property name="displayFormat">
       <string>yyyy-MM-dd HH:mm:ss</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QDateTimeEdit" name="endDateTimeEdit">
      <property name="dateTime">
       <datetime>
        <hour>1</hour>
        <minute>0</minute>
        <second>0</second>
        <year>2025</year>
        <month>5</month>
        <day>6</day>
       </datetime>
      </property>
      <property name="displayFormat">
       <string>yyyy-MM-dd HH:mm:ss</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="microservicesCountLabel">
      <property name="text">
       <string>Кількість мікросервісів:</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QSpinBox" name="microservicesCountSpinBox">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>1</number>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="2">
     <widget class="QPushButton" name="generateButton">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>Згенерувати часовий ряд</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0" colspan="2">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="6" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Побудова графіку</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Виберіть номер мікросервісу:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="microserviceNumberComboBox">
         <property name="currentIndex">
          <number>-1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="plotButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Побудувати графік</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>785</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Допомога</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSave">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Зберегти як...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Вихід</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>Про програму</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>chooseConfigRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>configPathLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>284</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chooseDatasetRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>chooseDatasetLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>493</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>605</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chooseDatasetRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>chooseDatasetButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>493</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>709</x>
     <y>90</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chooseConfigRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>chooseConfigButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
